### PR TITLE
Add date checks to pmcrypto

### PR DESCRIPTION
**Pull request by KAYLukas**

### CODE REVIEW

File: lib/key/check.js [hunk 1]
```diff
@@ -74,3 +75,16 @@ export default function keyCheck(info, email, expectEncrypted) {
 
     return info;
 }
+
+export function dateChecks([ key ], date = serverTime()) {
+    const keys = [ key, ...key.subKeys ];
+    if (!keys.every(({ keyPacket }) => keyPacket.created < date)) {
```
Comment (dhoko) [inline lib/key/check.js]: you can use some ;)

File: test/key/info.js [hunk 2]
```diff
@@ -30,7 +30,47 @@ JLdxTBJgqQoLePyHnb4LEPZItOUOMiHRBLUV8PSj/dxLEmYlmbPFOTnE62izRX4Q
...
+RLfkk/sT5Ay2LtjvPUmynkGvI3LFRkkqiA4buDiFOV/pqdV6RGIXEOcyMyPJ
+hufD30wViByeDdxpJsFOlVnw7D02WDNUHomhN2m8ogLJyFbfrG8/c54VmX6O
+J5Tj5Die5J5mSWCHbi9O0TBBgCysDxAr7blqK2xBO85q+Kp0809lPiS97iDq
+uokpJQHZjIvfQ5/9tx1946Tvo0RX0A26JfOO+J68XA==
+=MPBF
+-----END PGP PUBLIC KEY BLOCK-----`;
+
+    test('creation test', async () => {
+        const { dateError } = await pmcrypto.keyInfo(creationkey);
+        assert.strictEqual(dateError, 'The self certifications are created with illegal times');
```
Comment (dhoko) [inline test/key/info.js]: Can you add more tests ?

Comment (KAYLukas) [reply test/key/info.js]: @sanjanarajan Do you know how to generate these keys?

Comment (sanjanarajan) [reply test/key/info.js]: What keys? You are already testing a key with invalid times, right?

Comment (sanjanarajan) [reply test/key/info.js]: That is sufficient

Comment (dhoko) [reply test/key/info.js]: There are 3 diff errors, we need tests for them.

Comment (sanjanarajan) [reply test/key/info.js]: Maybe you could just add a test for a normal key to make sure it doesn't throw

Comment (dhoko) [reply test/key/info.js]: There is always a test for this case, but it doesn't test what's missing for errors.

**Pull request merged**